packages/whook-graphql/src/index.test.ts

```diff
@@ -0,0 +1,237 @@
+  const $autoload = jest.fn();
+
+  const API: OpenAPIV3.Document = {
+    openapi: '3.0.2',
+    info: {
+      version: '1.0.0',
+      title: 'Sample OpenAPI',
+      description: 'A sample OpenAPI file for testing purpose.',
+    },
+    paths: {
+      [getGraphQLDefinition.path]: {
+        [getGraphQLDefinition.method]: {
+          ...getGraphQLDefinition.operation,
+          security: [
+            {
+              bearerAuth: ['user'],
+            },
+          ],
+        },
+      },
+      [postGraphQLDefinition.path]: {
+        [postGraphQLDefinition.method]: {
+          ...postGraphQLDefinition.operation,
+          security: [
+            {
+              bearerAuth: ['user'],
+            },
+          ],
+        },
+      },
+    },
+    components: {
+      securitySchemes: {
+        bearerAuth: {
+          type: 'http',
+          description: 'Bearer authentication with a user API token',
+          scheme: 'bearer',
+        },
+        basicAuth: {
+          type: 'http',
+          description: 'Basic authentication of an API client',
+          scheme: 'basic',
+        },
+      },
+    },
+  };
+  const authentication = { check: jest.fn() };
+
+  let $instance;
+
+  async function prepareEnvironment() {
+    const $ = await basePrepareEnvironment();
+
+    $.register(
+      initializer(
+        {
+          name: '$autoload',
+          type: 'service',
+          options: { singleton: true },
+        },
+        async () => $autoload,
+      ),
+    );
+    $.register(constant('BASE_PATH', BASE_PATH));
+    $.register(constant('API', API));
+    $.register(constant('ENV', {}));
+    $.register(constant('NODE_ENV', 'test'));
+    $.register(constant('PORT', PORT));
+    $.register(constant('HOST', HOST));
+    $.register(constant('DEBUG_NODE_ENVS', []));
+    $.register(constant('NODE_ENVS', ['test']));
+    $.register(constant('MECHANISMS', [BEARER_MECHANISM, BASIC_MECHANISM]));
+    $.register(constant('logger', logger));
+    $.register(constant('time', time));
+    $.register(constant('debug', debug));
+    $.register(constant('GRAPHQL_OPTIONS', {}));
+    $.register(initGraphQL);
+    $.register(
+      initializer(
+        {
+          name: 'HANDLERS',
+          type: 'service',
+          inject: ['getGraphQL', 'postGraphQL'],
+          options: { singleton: true },
+        },
+        async services => services,
+      ),
+    );
+    $.register(constant('authentication', authentication));
+    [
+      wrapHandlerWithAuthorization(initGetGraphQL),
+      wrapHandlerWithAuthorization(initPostGraphQL),
+    ].forEach(handlerInitializer => $.register(handlerInitializer));
+
+    const fragment: WhookGraphQLFragmentService = {
+      typeDefs: gql`
+        type Query {
+          hello: String
+        }
+      `,
+      resolvers: {
+        Query: {
+          hello: () => 'Hello world!',
+        },
+      },
+    };
+    $.register(constant('graphQLFragments', [fragment]));
```

Comment:
@Oupsla  fragments would be register as a simple list like here (or maybe an autoloader)?